pkg/controller/mysqlcluster/mysqlcluster_controller_test.go

```diff
@@ -151,6 +151,9 @@ var _ = Describe("MysqlCluster controller", func() {
 			// updated
 			Eventually(requests, timeout).Should(Receive(Equal(expectedRequest)))
 
+			// an extra reconcile request may appear
+			Eventually(requests, timeout).Should(Or(Receive(Equal(expectedRequest)), Not(Receive(Equal(expectedRequest)))))
```

Comment:
As you can see from https://drone.presslabs.net/presslabs/mysql-operator/798/5 this does not fully solves the issue. You need to fully drain the `requests` channel because this line dequeue only one request. Sometimes you might get more than one request so basically you need to drain the channel and then test that no new reconcile request pops up (test for infinite loops).

This is the entire fixed code:
```go
	// Wait for initial reconciliation
	Eventually(requests, timeout).Should(Receive(Equal(expectedRequest)))
	// Wait for a second reconciliation triggered by components being created
	Eventually(requests, timeout).Should(Receive(Equal(expectedRequest)))
	// TODO: find out why sometimes we get extra reconciliation requests and remove this loop
	done := time.After(100 * time.Millisecond)
drain:
	for {
		select {
		case <-requests:
			continue
		case <-done:
			break drain
		}
	}
	// We need to make sure that the controller does not create infinite loops
	Consistently(requests).ShouldNot(Receive(Equal(expectedRequest)))
```